### ISSUE TITLE
Creating a subdomain has completely chewed up the httpd.conf file

### ISSUE THREAD

**Issue by pixel-paul** [virtualmin/virtualmin-gpl#277]
CentOS Linux 7.9.2009
Webmin version | 1.973
Virtualmin version | 6.16

I added the PHP 7.3 SCL library and then I created a new subdomain (on a relatively old server) and I have a very messed up httpd.conf file. By the looks of things, it has merged multiple Virtual Hosts together. For example:
```

<VirtualHost 192.168.0.22:80>
RemoveHandler .php
RemoveHandler .php5
RemoveHandler .php7.0
RemoveHandler .php7.1
RemoveHandler .php7.2
RemoveHandler .php7.3
php_admin_value engine Off
FcgidMaxRequestLen 1073741824
ErrorLog /var/log/virtualmin/domain_error_log
IPCCommTimeout 31
CustomLog /var/log/virtualmin/domain_access_log combined
ScriptAlias /cgi-bin/ /home/domain/cgi-bin/
DirectoryIndex index.html index.htm index.php index.php4 index.php5
<Directory /home/domain/public_html>
Options -Indexes +IncludesNOEXEC +SymLinksIfOwnerMatch +ExecCGI
allow from all
AllowOverride All Options=ExecCGI,Includes,IncludesNOEXEC,Indexes,MultiViews,SymLinksIfOwnerMatch
Require all granted
AddType application/x-httpd-php .php
AddHandler fcgid-script .php
AddHandler fcgid-script .php5
AddHandler fcgid-script .php7.0
AddHandler fcgid-script .php7.1
AddHandler fcgid-script .php7.2
FCGIWrapper /home/domain/fcgi-bin/php7.2.fcgi .php
FCGIWrapper /home/domain/fcgi-bin/php5.fcgi .php5
FCGIWrapper /home/domain/fcgi-bin/php7.0.fcgi .php7.0
FCGIWrapper /home/domain/fcgi-bin/php7.1.fcgi .php7.1
AddType application/x-httpd-php .php
AddHandler fcgid-script .php
AddHandler fcgid-script .php5
AddHandler fcgid-script .php7.0
AddHandler fcgid-script .php7.1
AddHandler fcgid-script .php7.2
AddHandler fcgid-script .php7.3
FCGIWrapper /home/domain/fcgi-bin/php7.2.fcgi .php
FCGIWrapper /home/domain/fcgi-bin/php5.fcgi .php5
FCGIWrapper /home/domain/fcgi-bin/php7.0.fcgi .php7.0
FCGIWrapper /home/domain/fcgi-bin/php7.1.fcgi .php7.1
FCGIWrapper /home/domain/fcgi-bin/php7.2.fcgi .php7.2
FCGIWrapper /home/domain/fcgi-bin/php7.3.fcgi .php7.3
RewriteRule ^(.*) https://domain:10000/ [R]
RemoveHandler .php
RemoveHandler .php5
RemoveHandler .php7.0
RemoveHandler .php7.1
RemoveHandler .php7.2
php_admin_value engine Off
FcgidMaxRequestLen 1073741824
AddType application/x-httpd-php .php
AddHandler fcgid-script .php
AddHandler fcgid-script .php5
AddHandler fcgid-script .php7.0
AddHandler fcgid-script .php7.1
AddHandler fcgid-script .php7.2
AddHandler fcgid-script .php7.3
FCGIWrapper /home/scl/fcgi-bin/php7.2.fcgi .php
FCGIWrapper /home/scl/fcgi-bin/php5.fcgi .php5
FCGIWrapper /home/scl/fcgi-bin/php7.0.fcgi .php7.0
FCGIWrapper /home/scl/fcgi-bin/php7.1.fcgi .php7.1
FCGIWrapper /home/scl/fcgi-bin/php7.2.fcgi .php7.2
FCGIWrapper /home/scl/fcgi-bin/php7.3.fcgi .php7.3
ErrorLog /var/log/virtualmin/domain_error_log
CustomLog /var/log/virtualmin/domain_access_log combined
ScriptAlias /cgi-bin/ /home/domain/cgi-bin/
DirectoryIndex index.html index.htm index.php index.php4 index.php5
<Directory /home/domain/public_html>
IPCCommTimeout 31
Options -Indexes +IncludesNOEXEC +SymLinksIfOwnerMatch +ExecCGI
allow from all
AllowOverride All Options=ExecCGI,Includes,IncludesNOEXEC,Indexes,MultiViews,SymLinksIfOwnerMatch
Require all granted
AddType application/x-httpd-php .php
RemoveHandler .php
RemoveHandler .php5
RemoveHandler .php7.0
RemoveHandler .php7.1
RemoveHandler .php7.2
RemoveHandler .php7.3
php_admin_value engine Off
FcgidMaxRequestLen 1073741824
IPCCommTimeout 31
FCGIWrapper /home/domain/fcgi-bin/php7.0.fcgi .php7.0
FCGIWrapper /home/domain/fcgi-bin/php7.1.fcgi .php7.1
FCGIWrapper /home/domain/fcgi-bin/php7.2.fcgi .php7.2
</Directory>
<Directory /home/domain/cgi-bin>
allow from all
AllowOverride All Options=ExecCGI,Includes,IncludesNOEXEC,Indexes,MultiViews,SymLinksIfOwnerMatch
Require all granted
</Directory>
RewriteEngine on
RewriteCond %{HTTP_HOST} =webmail.domain
RewriteRule ^(.*) https://domain:20000/ [R]
    AddType application/x-httpd-php .php
    AddHandler fcgid-script .php
    AddHandler fcgid-script .php5
    AddHandler fcgid-script .php7.0
    AddHandler fcgid-script .php7.1
    AddHandler fcgid-script .php7.2
    AddHandler fcgid-script .php7.3
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.2.fcgi .php
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php5.fcgi .php5
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.0.fcgi .php7.0
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.1.fcgi .php7.1
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.2.fcgi .php7.2
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.3.fcgi .php7.3
AddHandler fcgid-script .php7.2
AddHandler fcgid-script .php7.3
FCGIWrapper /home/scl/fcgi-bin/php7.2.fcgi .php
FCGIWrapper /home/scl/fcgi-bin/php5.fcgi .php5
FCGIWrapper /home/scl/fcgi-bin/php7.0.fcgi .php7.0
FCGIWrapper /home/scl/fcgi-bin/php7.1.fcgi .php7.1
FCGIWrapper /home/scl/fcgi-bin/php7.2.fcgi .php7.2
FCGIWrapper /home/scl/fcgi-bin/php7.3.fcgi .php7.3
ServerName scl.my.tld
AddType application/x-httpd-php .php
AddHandler fcgid-script .php
    RemoveHandler .php
    RemoveHandler .php5
    RemoveHandler .php7.0
    RemoveHandler .php7.1
    RemoveHandler .php7.2
    RemoveHandler .php7.3
    php_admin_value engine Off
    FcgidMaxRequestLen 1073741824
FCGIWrapper /home/scl/fcgi-bin/php7.0.fcgi .php7.0
    IPCCommTimeout 31
RemoveHandler .php
RemoveHandler .php5
RemoveHandler .php7.0
RemoveHandler .php7.1
RemoveHandler .php7.2
RemoveHandler .php7.3
php_admin_value engine Off
FcgidMaxRequestLen 1073741824
AddHandler fcgid-script .php
AddHandler fcgid-script .php5
AddHandler fcgid-script .php7.0
AddHandler fcgid-script .php7.1
AddHandler fcgid-script .php7.2
IPCCommTimeout 31
FCGIWrapper /home/scl/fcgi-bin/php7.2.fcgi .php
FCGIWrapper /home/scl/fcgi-bin/php5.fcgi .php5
RemoveHandler .php
RemoveHandler .php5
    AddType application/x-httpd-php .php
    AddHandler fcgid-script .php
    AddHandler fcgid-script .php5
    AddHandler fcgid-script .php7.0
    AddHandler fcgid-script .php7.1
    AddHandler fcgid-script .php7.2
    AddHandler fcgid-script .php7.3
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.2.fcgi .php
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php5.fcgi .php5
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.0.fcgi .php7.0
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.1.fcgi .php7.1
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.2.fcgi .php7.2
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.3.fcgi .php7.3
RewriteRule ^(.*) https://scl.my.tld:20000/ [R]
RewriteCond %{HTTP_HOST} =admin.scl.my.tld
RewriteRule ^(.*) https://scl.my.tld:10000/ [R]
    AddType application/x-httpd-php .php
    AddHandler fcgid-script .php
    AddHandler fcgid-script .php5
    AddHandler fcgid-script .php7.0
    AddHandler fcgid-script .php7.1
    AddHandler fcgid-script .php7.2
    AddHandler fcgid-script .php7.3
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.2.fcgi .php
    RemoveHandler .php
    RemoveHandler .php5
    RemoveHandler .php7.0
    RemoveHandler .php7.1
    RemoveHandler .php7.2
    RemoveHandler .php7.3
    php_admin_value engine Off
    FcgidMaxRequestLen 1073741824
AddHandler fcgid-script .php5
AddHandler fcgid-script .php7.0
AddHandler fcgid-script .php7.1
AddHandler fcgid-script .php7.2
AddHandler fcgid-script .php7.3
IPCCommTimeout 31
FCGIWrapper /home/scl/fcgi-bin/php7.2.fcgi .php
FCGIWrapper /home/scl/fcgi-bin/php5.fcgi .php5
FCGIWrapper /home/scl/fcgi-bin/php7.0.fcgi .php7.0
FCGIWrapper /home/scl/fcgi-bin/php7.1.fcgi .php7.1
    RemoveHandler .php
    RemoveHandler .php5
    RemoveHandler .php7.0
    RemoveHandler .php7.1
    RemoveHandler .php7.2
    RemoveHandler .php7.3
    php_admin_value engine Off
    FcgidMaxRequestLen 1073741824
AddHandler fcgid-script .php5
    IPCCommTimeout 31
AddHandler fcgid-script .php7.0
AddHandler fcgid-script .php7.1
AddHandler fcgid-script .php7.2
FCGIWrapper /home/scl/fcgi-bin/php7.2.fcgi .php
FCGIWrapper /home/scl/fcgi-bin/php5.fcgi .php5
RemoveHandler .php
RemoveHandler .php5
RemoveHandler .php7.0
RemoveHandler .php7.1
RemoveHandler .php7.2
RemoveHandler .php7.3
php_admin_value engine Off
FcgidMaxRequestLen 1073741824
Require all granted
</Directory>
RewriteEngine on
RewriteCond %{HTTP_HOST} =webmail.scl.my.tld
RewriteRule ^(.*) https://scl.my.tld:20000/ [R]
    AddType application/x-httpd-php .php
    AddHandler fcgid-script .php
    AddHandler fcgid-script .php5
    AddHandler fcgid-script .php7.0
    AddHandler fcgid-script .php7.1
    AddHandler fcgid-script .php7.2
    AddHandler fcgid-script .php7.3
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.2.fcgi .php
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php5.fcgi .php5
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.0.fcgi .php7.0
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.1.fcgi .php7.1
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.2.fcgi .php7.2
    FCGIWrapper /home/domain/domains/beta.domain/fcgi-bin/php7.3.fcgi .php7.3
SSLCertificateFile /home/my.tld/ssl.cert
SSLCertificateKeyFile /home/my.tld/ssl.key
SSLCACertificateFile /home/my.tld/ssl.ca
SSLProtocol all -SSLv2 -SSLv3 -TLSv1 -TLSv1.1
</VirtualHost>
```

**Comment by pixel-paul** [comment]
I was able to manually restore the httpd.conf file from yesterdays backup. At a guess, something didn't work correctly during the update of the httpd.conf file when I installed php 7.3. When I added the new domain, the regex went a little bit awry?

**Comment by swelljoe** [comment]
Was this actually a "Subdomain" or is it a Sub-Server? i.e. do you have the old subdomain site type enabled in Virtualmin?

**Comment by pixel-paul** [comment]
I had created a 'Sub-server'.

**Comment by jcameron** [comment]
Was this created using the Virtualmin UI, or the command-line API?

**Comment by pixel-paul** [comment]
It was created through the Virtualmin UI. I've looked into this further, and load on the server was running quite high when I ran 'check virtualmin configuration', which is when, I assume, php 7.3 would have been added to httpd.conf. I do recall having to run this twice at the time as the UI lost its connection. It was after this that I added the sub server, and it was created successfully. It was only when I tried to restart Apache that the errors in httpd.conf became apparent. On 15 Apr 2021 02:44, Jamie Cameron ***@***.***> wrote:
Was this created using the Virtualmin UI, or the command-line API?
